pkg/ingester/instance.go

```diff
@@ -374,10 +414,16 @@ func (i *instance) QuerySample(ctx context.Context, req logql.SelectSampleParams
 		shard = &shards[0]
 	}
 
+	var chunkFilter storage.ChunkFilterer
+	if i.chunkFilter != nil {
```

Comment:
Technically we should also apply filter for delete here, because someone could extract labels from the log line and access the data that should have been delete.

Not sure if this is MVP but I assume in the long term we will need to do this.

Reply:
Should'nt be too hard btw.